tests/func/test_import.py

```diff
@@ -156,3 +156,30 @@ def test(self):
             self.assertEqual(0, ret)
 
         self.assertEqual(1, remove_outs_call_counter.mock.call_count)
+
+
+class TestImportFilename(TestDvc):
+    def setUp(self):
+        super(TestImportFilename, self).setUp()
+        tmp_dir = self.mkdtemp()
+        self.external_source = os.path.join(tmp_dir, "file")
+        with open(self.external_source, "w") as fobj:
+            fobj.write("content")
+
+    def test(self):
+        ret = main(["import", "-f", "bar.dvc", self.external_source])
+        self.assertEqual(0, ret)
+        self.assertTrue(os.path.exists("bar.dvc"))
+
+        os.remove("bar.dvc")
+
+        ret = main(["import", "--file", "bar.dvc", self.external_source])
+        self.assertEqual(0, ret)
+        self.assertTrue(os.path.exists("bar.dvc"))
+
+        os.remove("bar.dvc")
+        os.mkdir("sub")
+
+        ret = main(["import", "--file", "sub/bar.dvc", self.external_source])
```

Comment:
Oh, one more thing. Please don't hardcode paths like `sub/bar.dvc` and use `os.path.join()` instead. We are running on both windows and *nix, so your test on windows was not creating `bar.dvc` in `sub` but rather a sole file named `sub/bar.dvc`(on windows / is a valid char in the filename).